emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py

```diff
@@ -109,3 +111,184 @@ def evaluate(self, x: np.ndarray) -> np.ndarray:
+            left = 2. ** i * np.min(fmean - 5. * fsd) + (1. - 2. ** i) * right
+            i += 1
+        i = 0
+        while probf(right) > 0.25:
+            right = -2. ** i * np.min(fmean - 5. * fsd) + (1. + 2. ** i) * fmean[idx].flatten()
+            i += 1
+
+        # Binary search for 3 percentiles
+        q1, med, q2 = map(lambda val: bisect(lambda x: probf(x) - val, left, right, maxiter=10000, xtol=0.00001),
+                            [0.25, 0.5, 0.75])
+
+        # solve for gumbel params
+        beta = (q1 - q2) / (np.log(np.log(4. / 3.)) - np.log(np.log(4.)))
+        alpha = med + beta * np.log(np.log(2.))
+
+        # sample K length vector from unif([0,1])
+        # return K Y* samples
+        self.mins = -np.log(-np.log(np.random.rand(self.num_samples))) * beta + alpha
+
+    def evaluate(self, x: np.ndarray) -> np.ndarray:
+        """
+        Computes the information gain, i.e the change in entropy of p_min if we would evaluate x.
+        :param x: points where the acquisition is evaluated.
+        """
+        if not self._required_parameters_initialized():
+            self.update_parameters()
+
+        # need model predictions at x and at the  x evaluated on the target fidelity (x_target_fidelity)
+        # remove current fidelity index from sample
+        x_target_fidelity = np.delete(x, self.source_idx, axis=1)
+        # Add target fidelity index to sample
+        idx = np.ones((x.shape[0])) * self.target_information_source_index
+        x_target_fidelity = np.insert(x_target_fidelity, self.source_idx, idx, axis=1)
+
+        # get predicted means and variances and pair-wise covariance
+        fmean, fvar = self.model.predict(x)
+        fsd = np.sqrt(fvar)
+        targetmean, targetvar = self.model.gpy_model.predict(x_target_fidelity, include_likelihood=False)
+        targetsd = np.sqrt(targetvar)
+        # faster to do for loop rather than vectorize to avoid unecessary between term covariance calculations
+        covs = [self.model.get_covariance_between_points(x[i].reshape(1, -1), x_target_fidelity[i].reshape(1, -1)) for i in range(0, x.shape[0])]
+        covs = np.array(covs).reshape(-1, 1)
+
+        # convert to pair-wise correlations
+        corrs = covs / (fsd*targetsd)
+        # correct for some potential rounding errors
+        corrs = np.clip(corrs, -1, 1)
+        corrs[x[:, -1] == self.source_idx] = 1
+
+        # calculate moments of extended skew Gaussian distributions
+        gammas = (self.mins-targetmean)/targetsd
+        denom = 1 / (1 - norm.cdf(gammas))
+        # need to account for numerical instability
+        denom[denom == np.inf] = 0
+        means = corrs * (norm.pdf(gammas)) * denom
+        Vars = 1 + corrs * (means) * (gammas - norm.pdf(gammas) * denom)
+        Vars[Vars <= 0] = 0
+
+        # get upper limits for numerical integration
+        upper_lim = means + 8 * np.sqrt(Vars)
+        lower_lim = means - 8 * np.sqrt(Vars)
```

Comment:
what's 8 here? why 8? ideally this shoud look like this:

    # comments explains why 8 is chosen
    number_of_std = 8
    upper_lim = means + whatever_8_represents  * np.sqrt(Vars)